.github/workflows/ci.yaml

```diff
@@ -13,6 +13,11 @@ on:
 
   workflow_dispatch:
 
+# Prevent concurrent runs on the same branch, except for the default branch.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.ref_type == 'branch' && github.ref_name == github.event.repository.default_branch && github.run_id }}
+  cancel-in-progress: true
```

Comment:
I have a counter-argument to ea5104f565a9ebbb0b9cc5e13ca69d760e22a522, describing a situation where concurrency is beneficial:

Sometimes, it's useful to show a difference in CI results between two commits. I have found myself pushing one commit immediately followed by another and relying on CI to run without interruption on both. With the proposed change, the same effect can only be accomplished in two ways:

1. Wait for the first commit's CI run to finish before pushing the next.
2. Temporarily remove this concurrency group in the PR.

I would rather cancel unwanted concurrent runs manually than use either of those approaches. Since ea5104f565a9ebbb0b9cc5e13ca69d760e22a522 is the only commit left in this PR, I'm just going to close it.